IdentityCore/src/util/mac/MSIDXpcProviderCaching.h

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDDeviceInfo.h"
+
+@class MSIDXpcConfiguration;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MSIDXpcProviderCaching <NSObject>
+
+// cachedXpcProvider is the Xpc provider's identifier from cache. This value can be updated through SsoExtension request
+@property (nonatomic) MSIDSsoProviderType cachedXpcProviderType;
+
+// cachedXpcStatus is the Xpc provider's status from cache.
+@property (nonatomic) BOOL cachedXpcStatus;
```

Comment:
nit: I wonder if this variable could be named in a more descriptive way. It might have been a tiny bit easier to follow if this was named someting like "cachedCanPerformRequestsStatus" or something to know that it is the cached response value for canPerformRequests

Reply:
Updated the naming